azure-iot-device/azure/iot/device/provisioning/internal/request_response_provider.py

```diff
@@ -0,0 +1,80 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+import json
+from ..models.registration_result import RegistrationResult, RegistrationState
+
+logger = logging.getLogger(__name__)
+
+
+class RequestResponseMachine(object):
+    def __init__(self, state_based_provider):
+
+        self._state_based_provider = state_based_provider
+        self._state_based_provider.on_state_based_provider_message_received = self._receive_response
+
+        self.on_response_received = None
+
+    def send_request(self, publish_action):
+        """
+        :param publish_action: The publish action
+        """
+        self.publish([publish_action])
+
+    def connect(self, callback=None, post_connect_actions=None):
+        # TODO : MQTT Transport connect should take post connect actions.
```

Comment:
># TODO : MQTT Transport connect should take post connect actions. [](start = 8, length = 65)

Yes, absolutely!

Reply:
this was a reminder for the whoever would be doing transport but based on the comment on line 22 does the action get created in this class to be passed along to the mqtt transport (the state based provider)

---
In reply to: [277396387](https://github.com/Azure/azure-iot-sdk-python-preview/pull/81#discussion_r277396387) [](ancestors = 277396387)